db/db_impl/db_impl.cc

```diff
@@ -1470,8 +1476,9 @@ Status DBImpl::MarkLogsSynced(uint64_t up_to, bool synced_dir) {
     assert(wal.getting_synced);
     if (immutable_db_options_.track_and_verify_wals_in_manifest &&
         wal.writer->file()->GetFileSize() > 0) {
-      synced_wals.AddWal(wal.number,
-                         WalMetadata(wal.writer->file()->GetFileSize()));
+      uint64_t tracked_synced_size = std::min(
+          wal.tracked_synced_size_limit, wal.writer->file()->GetFileSize());
```

Comment:
Should we have a unit test for the case in which `wal.tracked_synced_size_limit > wal.writer->file()->GetFileSize()` when tracked_synced_size_limit != U64_MAX, or is it possible?

Reply:
Not sure if possible. I have been assuming `wal.writer->file()->GetFileSize()` is monotonically increasing in which case it would not be possible (when `tracked_synced_size_limit` is set, it comes from an earlier `GetFileSize()`), but perhaps recycling changes things. Let me check.

Reply:
Looks impossible. A `WritableFileWriter` only keeps track of the size written by itself. I added an assertion now that `pre_sync_size`, which is assigned from `GetFileSize()`, is monotonically increasing.